### ISSUE TITLE
docs: Explain port-specific differences in network / WLAN / ssl

### ISSUE THREAD

**Issue by jimmo** [micropython/micropython#5245]
Inspired by https://forum.micropython.org/viewtopic.php?f=18&t=7119

At a minimum, just add the auth mode constants to network.rst. (And highlight if only some are supported by a subset of ports).

It's worth doing a bit of a pass through for related improvements. There are a few examples like that where it can be difficult to figure out how to use a feature. Most features should be supported on all ports, so hopefully this is fairly straightforward.

Related with ssl, one question that pops up a lot is how to use cert validation, where the docs says  "Depending on the underlying module implementation in a particular MicroPython port, some or all keyword arguments above may be not supported." So in a similar vein, would be good to address this too in the same way.

**Comment by dpgeorge** [comment]
> Related with ssl, one question that pops up a lot is how to use cert validation,

Yes, would be good to make examples for this.  Eg the existing `examples/network/http_server_ssl.py` is a good candidate to explain certs at the server side and needs updating because it doesn't currently work with CPython.
